webapp/src/containers/shareRecords/index.tsx

```diff
@@ -0,0 +1,175 @@
+            kb_knowledge_base: recordData?.kb_knowledge_base || '',
+            workflow_state: recordData?.workflow_state || '',
+        };
+
+        setShareRecordPayload(payload);
+        makeApiRequest(Constants.pluginApiServiceConfigs.shareRecord.apiServiceName, payload);
+    };
+
+    useEffect(() => {
+        if (channel || !suggestionChosen) {
+            setError(false);
+        }
+
+        if (!suggestionChosen) {
+            setChannel(null);
+        }
+    }, [channel, suggestionChosen]);
+
+    return (
+        <Modal
+            show={pluginState.openShareRecordModalReducer.open}
+            onHide={hideModal}
+            className={'rhs-modal'}
+        >
+            <>
+                <ModalHeader
+                    title={'Share a record'}
+                    onHide={hideModal}
+                    showCloseIconInHeader={true}
+                />
+                <RecordTypePanel
+                    recordType={recordType}
+                    setRecordType={setRecordType}
+                    setResetRecordPanelStates={setResetRecordPanelStates}
+                    placeholder={'Record Type'}
+                    recordTypeOptions={Constants.shareRecordTypeOptions}
+                />
+                <SearchRecordsPanel
+                    recordValue={recordValue}
+                    setRecordValue={setRecordValue}
+                    suggestionChosen={suggestionChosen}
+                    setSuggestionChosen={setSuggestionChosen}
+                    recordType={recordType}
+                    setApiError={setApiError}
+                    setShowModalLoader={setShowModalLoader}
+                    recordId={recordId}
+                    setRecordId={setRecordId}
+                    resetStates={resetRecordPanelStates}
+                    setResetStates={setResetRecordPanelStates}
+                    setRecordData={setRecordData}
+                    disabled={!recordType}
+                />
+                {suggestionChosen && (
+                    <ChannelPanel
+                        channel={channel}
+                        setChannel={setChannel}
+                        setShowModalLoader={setShowModalLoader}
+                        setApiError={setApiError}
+                        channelOptions={channelOptions}
+                        setChannelOptions={setChannelOptions}
+                        actionBtnDisabled={showModalLoader}
+                        placeholder={'Search channel to share'}
+                        validationError={error}
+                    />
+                )}
+                <ModalSubtitleAndError error={apiError ?? ''}/>
+                <ModalFooter
+                    onConfirm={suggestionChosen ? shareRecord : null}
+                    confirmBtnText={'Share'}
+                    confirmDisabled={showModalLoader}
+                    onHide={hideModal}
+                    cancelBtnText={'Cancel'}
```

Comment:
Fix this at all such places.

```suggestion
                    cancelBtnText='Cancel'
```